### DISCUSSION TITLE
Export to SD card option?

### DISCUSSION THREAD

**Discussion by ghost** [iSoron/uhabits#536] [Feature Requests]
Am I missing this option? When I click export I get a list of share items such as text or email. But I would like to back up my data to the local SD card instead.

**Comment by iSoron** [comment]
@evitalis Every time you export your backup, a copy is automatically saved in your internal storage (Internal Storage » Android » data » org.isoron.uhabits » files » Backups). Be aware that this folder will be deleted when the app is uninstalled. If you would like to keep the backups on the SD card after the app is uninstalled, you should move them to another folder.

**Reply by alphanerdgen** [reply]
Files and folders within Android > Data are no longer accessible in Android 11 thanks to Google's new storage policy.

Sending via Bluetooth isn't working either. Maybe because it doesn't recognise .db files.

Forced to upload the backup to Signal and then download it. This isn't convenient at all.

**Comment by sameerhussain** [comment]
Can we export data in excel sheet format to SD card and import it same way back from SD card?

**Comment by ghost** [comment]
@iSoron I am looking for a way to bypass any of the other export methods and only export to the SD card.

**Comment by szpak** [comment]
@evitalis It's just a workaround, but having an application such as ($free and without ads) [Total Commander](https://play.google.com/store/apps/details?id=com.ghisler.android.TotalCommander&hl=en) allows you to handle that sharing request and just save the backup file to the SD card (or some other folder in the internal storage).

**Comment by iSoron** [comment]
> @iSoron I am looking for a way to bypass any of the other export methods and only export to the SD card.

Currently, you can click "Export full backup" then close the dialog that asks you to pick an app. The backup will still be saved to the SD card. I'm aware this is not intuitive at all; we should improve this.

**Comment by iSoron** [comment]
Just to expand on this issue a little bit, the reason it's so awkward to export to the SD card is because (i) we don't have permission to see other SD card folders, only the ridiculously hidden `Android/data/org.isoron.uhabits/...`; and (ii) even if we could ask permission to read the entire SD card, this option will likely disappear in a future Android version. Google has been slowly but persistently pushing developers to use the Storage Access Framework instead; and Loop already uses it.

In theory, after Loop generates a file, the user should be able to pick any file manager they already have installed, and the file manager should provide the option to save the file anywhere (either in the cloud or in the internal storage). In practice, the default file managers are completely unintuitive. The SD card option, if present at all, is typically hidden.

**Comment by lldata** [comment]
Maybe this problem can be solved with the PR related to #563 as well. 

At least with my PR I was able to export data as a CSV zip file to my google drive and reimport it again (tested in emulator). 

For me personally I'd much prefer having a backup in the cloud, rather than some obscure folder on a sd card.
